accessible/toolboxview.component.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a toolbox is rendered
+ * in AccessibleBlockly. Also handles any interactions with the toolbox.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.ToolboxView = ng.core
+.Component({
+  selector: 'toolbox-view',
+  template: `
+    <h3 #toolboxTitle id="blockly-toolbox-title">Toolbox</h3>
+    <ol #tree id="blockly-toolbox-tree" *ngIf="makeArray(sightedToolbox) && makeArray(sightedToolbox).length > 0" tabIndex="0" role="group" class="blocklyTree" [attr.aria-labelledby]="toolboxTitle.id" (keydown)="treeService.keyHandler($event, tree)">
```

Comment:
It seems like makeArray and sightedToolbox always go together, which makes me wonder if you should just have a single  getToolboxCategories() function instead.